src/Renode/Network/NetworkServer/NetworkServer.cs

```diff
@@ -218,4 +245,76 @@ private bool TryHandleArp(ARPPacket packet, out ARPPacket response)
         private readonly Dictionary<string, int> modulesNames;
         private readonly Dictionary<IPAddress, PhysicalAddress> arpTable;
     }
+
+    public class IcmpServer: IExternal
+    {
+
+        public IcmpServer(IPAddress serverIP, MACAddress serverMAC)
+        {
+            IP = serverIP;
+            MAC = serverMAC;
+            arpTable = new Dictionary<IPAddress, PhysicalAddress>();
+        }
+        
+        public MACAddress MAC { get; set; }
+        public IPAddress IP { get; set; }
+        public Dictionary<IPAddress, PhysicalAddress> arpTable;
+
+        public void HandleIcmpPacket(Action<EthernetFrame> FrameReady, IPv4Packet packet)
```

Comment:
add destinationMac as a parameter

Reply:
maybe we should use `ICMPv4Packet` as a parameter? `HandleUDP` gets the `UDPPacket`